caddy.go

```diff
@@ -253,12 +243,24 @@ type Server interface {
 	// connections.
 	Listen() (net.Listener, error)
 
+	// ListenPacket starts listening by creating a new packetconn
+	// and returning it. It does not start accepting
+	// connections.
+	ListenPacket() (net.PacketConn, error)
+
 	// Serve starts serving using the provided listener.
 	// Serve must start the server loop nearly immediately,
 	// or at least not return any errors before the server
 	// loop begins. Serve blocks indefinitely, or in other
 	// words, until the server is stopped.
 	Serve(net.Listener) error
+
+	// ServePacket starts serving using the provided packetconn.
+	// Serve must start the server loop nearly immediately,
+	// or at least not return any errors before the server
+	// loop begins. Serve blocks indefinitely, or in other
+	// words, until the server is stopped.
+	ServePacket(net.PacketConn) error
```

Comment:
Change instances of `Serve` to `ServePacket`. And again, explain when this should be implemented vs. no-op; if only TCP is used, maybe only Listen() and Serve() need to be implemented? Should ServePacket always be implemented if ListenPacket is (i.e. do they pair up)? etc.